tooling/minimize/src/util.rs

```diff
@@ -0,0 +1,132 @@
+    if std::env::var("MINIMIZE_BE_RUSTC").as_deref() != Ok("sysroot") {
+        let sysroot = get_sysroot_dir();
+        args.insert(index, format!("--sysroot={}", sysroot.display()));
+    }
+}
+
+pub fn show_error(msg: &impl std::fmt::Display) -> ! {
+    eprintln!("fatal error: {msg}");
+    std::process::exit(101) // exit code needed to make ui_test happy
+}
+
+#[macro_export]
+macro_rules! show_error {
+    ($($tt:tt)*) => {show_error(&format_args!($($tt)*)) };
+}
+
+pub fn get_sysroot_dir() -> PathBuf {
+    match std::env::var_os("MINIRUST_SYSROOT") {
+        Some(dir) => PathBuf::from(dir),
+        None => {
+            let user_dirs = directories::ProjectDirs::from("org", "rust-lang", "minirust").unwrap();
+            user_dirs.cache_dir().to_owned()
+        }
+    }
+}
+
+pub fn be_rustc() {
+    // Get the rest of the command line arguments
+    let mut args: Vec<String> = env::args().skip(1).collect();
+
+    let use_panic_abort = args
+        .array_windows::<2>()
+        .any(|[first, second]| first == "--crate-name" && second == "panic_abort");
+
+    insert_default_args(&mut args, 0);
+
+    if use_panic_abort {
+        args.insert(0, "-Cpanic=abort".into());
+    } else {
+        args.insert(0, "-Cpanic=unwind".into());
+    }
+
+    // Invoke the rust compiler
+    let status = Command::new("rustc")
+        .args(args)
+        .env_remove("RUSTC")
+        .env_remove("MINIMIZE_BE_RUSTC")
+        .status()
+        .expect("failed to invoke rustc in custom sysroot build");
+
+    std::process::exit(status.code().unwrap_or(1));
+}
+
+// Builds (if necessary) and returns a path to the sysroot for our custom MIR compiled libraries
+pub fn setup_sysroot() -> PathBuf {
+    // Determine where to put the sysroot.
+    let sysroot_dir = get_sysroot_dir();
+    let sysroot_dir = sysroot_dir.canonicalize().unwrap_or(sysroot_dir); // Absolute path 
+
+    // Determine where the rust sources are located.
+    let rust_src = {
+        // Check for `rust-src` rustup component.
+        let rustup_src = rustc_build_sysroot::rustc_sysroot_src(Command::new("rustc"))
+            .expect("could not determine sysroot source directory");
+        if !rustup_src.exists() {
+            show_error!("`rust-src` not found");
+        } else {
+            rustup_src
+        }
+    };
+    if !rust_src.exists() {
+        show_error!("given Rust source directory `{}` does not exist.", rust_src.display());
+    }
+    if rust_src.file_name().and_then(OsStr::to_str) != Some("library") {
+        show_error!(
+            "given Rust source directory `{}` does not seem to be the `library` subdirectory of \
+             a Rust source checkout.",
+            rust_src.display()
+        );
+    }
+
+    let mut it = std::env::args();
+    let target = it
+        .by_ref()
+        .position(|a| a == "--target")
+        .and_then(|_| it.next())
+        .unwrap_or_else(|| rustc_version::version_meta().expect("rustc").host);
```

Comment:
This is somewhat odd, you are calling `position` and then ignoring the result and relying on the side-effect this had on `it`... this should be doable in a nicer way with `array_windows` and `find_map`.